core/common/src/main/java/alluxio/security/group/GroupMappingService.java

```diff
@@ -37,27 +37,68 @@
   class Factory {
     private static final Logger LOG = LoggerFactory.getLogger(Constants.LOGGER_TYPE);
 
+    // TODO(chaomin): maintain a map from SECURITY_GROUP_MAPPING_CLASS name to cachedGroupMapping.
+    // Currently the single global cached GroupMappingService assumes that there is no dynamic
+    // configuration change for {@link Constants#SECURITY_GROUP_MAPPING_CALSS}.
+    private static GroupMappingService sGroupMappingService = null;
+    private static CachedGroupMapping sCachedGroupMapping = null;
+
     // prevent instantiation
     private Factory() {}
 
     /**
-     * Gets the groups being used to map user-to-groups.
+     * Gets the groups mapping service being used to map user-to-groups.
      *
-     * @return the groups being used to map user-to-groups
+     * @return the groups mapping service being used to map user-to-groups
      */
-    public static GroupMappingService getUserToGroupsMappingService() {
-      GroupMappingService mGroupMappingService;
+    public static GroupMappingService get() {
       if (LOG.isDebugEnabled()) {
         LOG.debug("Creating new Groups object");
       }
       try {
-        mGroupMappingService = CommonUtils.createNewClassInstance(
-            Configuration.<GroupMappingService>getClass(Constants.SECURITY_GROUP_MAPPING), null,
-            null);
+        if (sGroupMappingService == null) {
```

Comment:
I guess I am asking why we need to sync here rather than initializing in the static block. later I realize that it is probably due to an effort to avoid make static init block depends on `Constant` or `Configuration`. I guess this can be solved later with my configuration change